src/trace-agent.js

```diff
@@ -35,8 +36,10 @@ var traceAgent;
 function TraceAgent(config, logger) {
   this.config_ = config;
   this.logger = logger;
+  this.plugins = config && config.plugins ? config.plugins : {};
```

Comment:
We should normalize config when we received it from the user, and assume in the rest of the code that all expected properties are defined.